par_test.go

```diff
@@ -314,87 +313,145 @@ func BenchmarkReduce(b *testing.B) {
 	})
 }
 
-var result bool
-
 func BenchmarkAny(b *testing.B) {
-	rand.Seed(time.Now().Unix())
+	rand.Seed(1)
 	collections := CreateCollections(10000)
-	maxSum := CollectionsGetMaxSum(collections)
 
-	b.Run("serial", func(b *testing.B) {
-		var r bool
+	b.Run("serial with match", func(b *testing.B) {
 		for n := 0; n < b.N; n++ {
+			needle := collections[len(collections)*n/b.N].NumbersSum()
+			var result bool
 			for _, c := range collections {
-				if c.NumbersSum() == maxSum {
-					r = true
+				if c.NumbersSum() == needle {
+					result = true
 					break
 				}
 			}
+			_ = result
```

Comment:
Does this actually prevent the compiler from optimising out the benchmark?

I've always used the global to prevent that as [advised here](https://dave.cheney.net/2013/06/30/how-to-write-benchmarks-in-go)

Reply:
TBH at some point in the past I noticed that I could swap assigning to a global with assigning to `_` prevent benchmarks from getting optimized out, but I just ran some simple optimization pass tests and assigning to `_` does indeed get optimized out, but I'm failing to reproduce this in benchmarks, i.e. looks like assignment to anything is no longer necessary inside benchmarks.

However, that seems like an unstable assumption to make, so just to remove any doubt, I added result-based assignment to a global into all of the benchmarks. :+1: